internal/overlord/servstate/manager.go

```diff
@@ -327,3 +342,52 @@ func (m *ServiceManager) ServiceLogs(services []string, last int) (map[string]se
 
 	return iterators, nil
 }
+
+// Replan returns a list of services to stop and services to start because
+// their plans had changed between when they started and this call.
+func (m *serviceManager) Replan() ([]string, []string, error) {
+	releasePlan, err := m.acquirePlan()
+	if err != nil {
+		return nil, nil, err
+	}
+	defer releasePlan()
+
+	needsRestart := make(map[string]bool)
+	stopOrder := []string(nil)
```

Comment:
Nit: in my experience `var stopOrder []string` is more common for the zero (nil) value. Not sure exactly why, but it is (maybe because there's less punctuation?). Same below.

For example, in Juju there are 1371 instances of `var foo []T`, but only 3 of `foo := []T(nil)`. Similar in the Go stdlib.